tests/store/artifact/test_models_artifact_repo.py

```diff
@@ -111,50 +111,104 @@ def test_models_artifact_repo_init_with_stage_uri_and_not_using_databricks_regis
         get_repo_mock.assert_called_once_with(artifact_location)
 
 
-def test_models_artifact_repo_uses_repo_download_artifacts():
+def test_models_artifact_repo_uses_repo_download_artifacts(tmp_path):
     """
-    ``ModelsArtifactRepository`` should delegate `download_artifacts` to its
-    ``self.repo.download_artifacts`` function.
+    `ModelsArtifactRepository` should delegate `download_artifacts` to its
+    `self.repo.download_artifacts` function.
     """
     artifact_location = "s3://blah_bucket/"
+    dummy_file = tmp_path / "dummy_file.txt"
+    dummy_file.write_text("dummy content")
+
     with mock.patch.object(
         MlflowClient, "get_model_version_download_uri", return_value=artifact_location
     ), mock.patch.object(ModelsArtifactRepository, "_add_registered_model_meta_file"):
         model_uri = "models:/MyModel/12"
         models_repo = ModelsArtifactRepository(model_uri)
-        models_repo.repo = Mock()
-        models_repo.download_artifacts("artifact_path", "dst_path")
-        models_repo.repo.download_artifacts.assert_called_once()
+        models_repo.repo = mock.Mock(**{"download_artifacts.return_value": str(dummy_file)})
+
+        models_repo.download_artifacts("artifact_path", str(tmp_path))
+
+        models_repo.repo.download_artifacts.assert_called_once_with("artifact_path", str(tmp_path))
+
 
+@pytest.mark.skipif(is_windows(), reason="This test fails on Windows")
+def test_models_artifact_repo_download_with_real_files(tmp_path):
+    # Simulate an artifact repository
+    temp_remote_storage = tmp_path / "remote_storage"
+    model_dir = temp_remote_storage / "model_dir"
+    model_dir.mkdir(parents=True)
+    mlmodel_path = model_dir / "MLmodel"
+    mlmodel_path.write_text("dummy content")
```

Comment:
```suggestion
    mlmodel_path.touch()
```

we can use more `touch`.